### ISSUE TITLE
a2a3: VecTile [16,1] column load from GlobalTensor fails TLOAD layout check

### ISSUE THREAD

**Issue by high-cloud** [hw-native-sys/pypto#1398]
## Summary

Loading a column-shaped Vec tile (`[16, 1]`) from a row-major/ND global tensor fails during a2a3 incore compilation with:

```text
TLOAD(VecTile, GlobalTensor) only support ND2ND/DN2DN/NZ2NZ
```

This blocks T-axis tiled kernels that need to load a per-row scale column and broadcast it across a wider tile, such as `hc_pre.mix_x` using `[T_TILE, D_CHUNK]` tiles.

## Minimal reproducer

Complete standalone reproducer:

```python
# Copyright (c) PyPTO Contributors.
# This program is free software, you can redistribute it and/or modify it under the terms and conditions of
# CANN Open Software License Agreement Version 2.0 (the "License").
# Please refer to the License for details. You may not use this file except in compliance with the License.
# THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED,
# INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
# See LICENSE in the root of the software repository for the full text of the License.
# -----------------------------------------------------------------------------------------------------------
"""Reproducer for a2a3 VecTile column-load lowering.

The kernel loads ``scale[:, 0:1]`` as a ``[16, 1]`` Vec tile from a global
``[16, 8]`` tensor, then broadcasts it across ``x`` with ``row_expand_mul``.
On a2a3 this currently fails during incore compilation with:

    TLOAD(VecTile, GlobalTensor) only support ND2ND/DN2DN/NZ2NZ
"""

import pypto.language as pl

ROWS = 16
COLS = 64
SCALE_COLS = 8


def build_repro_program():
    @pl.program
    class VecColumnLoadRepro:
        @pl.function(type=pl.FunctionType.Opaque)
        def repro(
            self,
            x: pl.Tensor[[ROWS, COLS], pl.FP32],
            scale: pl.Tensor[[ROWS, SCALE_COLS], pl.FP32],
            y: pl.Out[pl.Tensor[[ROWS, COLS], pl.FP32]],
        ) -> pl.Tensor[[ROWS, COLS], pl.FP32]:
            with pl.at(level=pl.Level.CORE_GROUP, name_hint="vec_col_load"):
                col = pl.load(
                    scale,
                    [0, 0],
                    [ROWS, 1],
                    target_memory=pl.MemorySpace.Vec,
                )
                x_tile = pl.load(
                    x,
                    [0, 0],
                    [ROWS, COLS],
                    target_memory=pl.MemorySpace.Vec,
                )
                y_tile = pl.row_expand_mul(x_tile, col)
                y = pl.store(y_tile, [0, 0], y)
            return y

    return VecColumnLoadRepro


def build_specs():
    import torch
    from golden import TensorSpec

    return [
        TensorSpec("x", [ROWS, COLS], torch.float32, init_value=torch.randn),
        TensorSpec("scale", [ROWS, SCALE_COLS], torch.float32, init_value=torch.randn),
        TensorSpec("y", [ROWS, COLS], torch.float32, is_output=True),
    ]


def golden(values):
    values["y"][:] = values["x"] * values["scale"][:, 0:1]


if __name__ == "__main__":
    import argparse

    from golden import RunConfig, run

    parser = argparse.ArgumentParser()
    parser.add_argument("-p", "--platform", type=str, default="a2a3",
                        choices=["a2a3", "a2a3sim", "a5", "a5sim"])
    parser.add_argument("-d", "--device", type=int, default=0)
    args = parser.parse_args()

    result = run(
        program=build_repro_program(),
        specs=build_specs(),
        golden_fn=golden,
        config=RunConfig(
            rtol=1e-5,
            atol=1e-5,
            compile=dict(dump_passes=True),
            runtime=dict(platform=args.platform, device_id=args.device),
        ),
    )
    if not result.passed:
        if result.error:
            print(result.error)
        raise SystemExit(1)
```

## Reproduction command

Run through the NPU task queue:

```bash
task-submit --device 11 \
  --env PYTHONPATH --env ASCEND_HOME_PATH --env PTOAS_ROOT --env PTO_ISA_ROOT \
  --max-time 120 \
  --run "cd /data/yangyaodong/code/pto_workspace/modules/pypto-lib && python examples/bugs/repro_vec_col_load.py -p a2a3 -d {}"
```

Task used while reproducing:

```text
task_20260518_192628_356422018233
```

## Observed result

The PyPTO compile stage completes, then runtime-side incore compilation fails:

```text
[RUN] compile done (0.60s)
[RUN] runtime ...
[Incore] Compilation failed:
.../TLoad.hpp:459:9: error: static assertion failed due to requirement 'isSameLayout': Fix: TLOAD(VecTile, GlobalTensor) only support ND2ND/DN2DN/NZ2NZ!
...
TLOAD_TILE_IMPL<pto::Tile<pto::TileType::Vec, float, 16, 1, pto::BLayout::ColMajor, ...>,
                 pto::GlobalTensor<float, pto::Shape<1, 1, 1, 16, 1>, pto::Stride<128, 128, 128, 8, 1>, pto::Layout::ND>>
...
TLOAD(v11, v15);
```

## Expected result

One of:

- PyPTO lowers `[16, 1]` Vec column loads from ND global tensors to a legal a2a3 load sequence, or
- PyPTO rejects this shape/layout earlier with an actionable diagnostic and suggests the supported workaround.

## Notes

A related `hc_pre.mix_x` optimization wants to load `pre_val[t0:t0+T_TILE, h:h+1]` and use it as a per-row scale for a `[T_TILE, D_CHUNK]` tile. Direct `[TILE,1]` loading hits this compile error. Loading a wider `[TILE, HC_PAD]` tile avoids this exact compile error but led to runtime timeout in the larger kernel, so a legal direct column-load lowering would be useful.
